programs/uxd/tests/integration_tests/api/program_uxd/instructions/process_edit_controller.rs

```diff
@@ -0,0 +1,38 @@
+use anchor_lang::InstructionData;
+use anchor_lang::ToAccountMetas;
+use solana_program::instruction::Instruction;
+use solana_program_test::ProgramTestContext;
+use solana_sdk::signature::Keypair;
+use solana_sdk::signer::Signer;
+
+use crate::integration_tests::api::program_test_context;
+use crate::integration_tests::api::program_uxd;
+
+pub async fn process_edit_controller(
+    program_test_context: &mut ProgramTestContext,
+    program_keys: &program_uxd::accounts::ProgramKeys,
+    payer: &Keypair,
+    redeemable_global_supply_cap: Option<u128>,
+) -> Result<(), String> {
+    let accounts = uxd::accounts::EditController {
+        authority: program_keys.authority.pubkey(),
+        controller: program_keys.controller,
+    };
+    let payload = uxd::instruction::EditController {
+        fields: uxd::instructions::EditControllerFields {
+            redeemable_global_supply_cap,
+        },
+    };
+    let instruction = Instruction {
```

Comment:
Would be interesting to create a `utils` function to wrap the ix creation and execution

Reply:
Im a bit confused about this comment, I thought that was the `process_instruction `util already ? I think that would be difficult to make anything more generic than `process_instruction`

Reply:
You can wrap instruction & process_instruction

something like:

```
fn my_uxd_func(accounts, payload, payer, signer) { 
     program_id: uxd::id(),
        accounts: accounts.to_account_metas(None),
        data: payload.data(),
    };
    program_test_context::process_instruction_with_signer(
        program_test_context,
        instruction,
        payer,
        &signer
    )
    .await
}
```

Reply:
Ah I see, yeah that make sense, a bit overkill imo, since it only saves 3 LOC and only works for anchor calls (deosnt work for solana SPL for example) also I would need to make 2 variants for with/without signers